central/activecomponent/datastore/singleton.go

```diff
@@ -15,7 +15,7 @@ var (
 
 func initialize() {
 	storage := pgStore.New(globaldb.GetPostgres())
-	searcher := search.NewV2(storage, pgStore.NewIndexer(globaldb.GetPostgres()))
+	searcher := search.NewV2(storage)
 	ds = New(storage, searcher)
```

Comment:
Moving datastore interfaces to use the postgres interface rather than the store interface is a tight coupling. Many unit tests are built around validating the datastores with mock stores (or wrapper stores that wrap auto-generated stores) which gives us tremendous flexibility to test the datastore without requiring a live postgres database and enables us to work with a fixed behavior of the store, which is in fact mostly auto-generated.

That said, I firmly believe that this topic, although needs discussion, does not belong to the scope of this PR. Perhaps, the discussion can continue at other forums.